### DISCUSSION TITLE
Defective HM-Sec-RHS window sensor results in entire heater group shown as 'unavailable' in HA

### DISCUSSION THREAD

**Discussion by markuslang79** [SukramJ/aiohomematic#1882] [Q&A] [answered]
Hello,

I run several heater groups with older Homematic (non-IP) devices where in two groups some HM-Sec-RHS window sensors are defective due to the well known hardware issues of those sensors. This results in HA using the _custom_homematic_ integration that the entire heater group gets flagged as unavailable. After a CCU3 restart where there are no service messages about communication errors with those window sensors the heater group is available in HA and can be controlled accordingly. As soon as the service message pops up in the CCU3 the windows sensor(s) as well as the heater group itself gets flagged as unavailable in HA so that I'm pretty sure those defective window sensors are the root cause for this behavior. Also I can rule out any other general (setup) issue since the other heater groups without defective sensors work fine all the time.

Before creating any bug issue regarding this I wanted to ask if this is the expected behavior in such a situation or if this should be considered as a bug. I personally think that a communication error with a sensor shouldn't result in an unavailable/unusable heater group in HA since the heater group still works as it should on CCU3 level - except that some window sensors don't report an _window open_ state any longer.

I can provide further details, log files, etc in case this is something worthy to follow up.

Greetings,
Markus

**Comment by SukramJ** [comment, marked as answer]
This integration does not care about group members. If the group reports UNREACH then it's marked unavailable in HA.
Read [this](https://github.com/danielperna84/custom_homematic#homematicip_localforce_device_availability) in the docs.

The service is also used in the Reactivate* [blueprints](https://github.com/danielperna84/custom_homematic#available-blueprints).
